x-pack/plugins/security_solution/public/detections/components/alerts_table/alerts_grouping.tsx

```diff
@@ -125,153 +89,146 @@ export const GroupedAlertsTableComponent: React.FC<AlertsTableComponentProps> =
-  } = useQueryAlerts<
-    {},
-    GroupingAggregation<AlertsGroupingAggregation> &
-      GroupingFieldTotalAggregation<AlertsGroupingAggregation>
-  >({
-    query: queryGroups,
-    indexName: signalIndexName,
-    queryName: ALERTS_QUERY_NAMES.ALERTS_GROUPING,
-    skip: isNoneGroup(selectedGroup),
-  });
+  const resetAllPagination = useCallback(() => {
+    setPageIndex((curr) => curr.map(() => DEFAULT_PAGE_INDEX));
+  }, []);
 
   useEffect(() => {
-    if (!isNoneGroup(selectedGroup)) {
-      setAlertsQuery(queryGroups);
-    }
-  }, [queryGroups, selectedGroup, setAlertsQuery]);
+    resetAllPagination();
+  }, [resetAllPagination, selectedGroups]);
+
+  const setPageVar = useCallback(
+    (newNumber: number, groupingLevel: number, pageType: 'index' | 'size') => {
+      if (pageType === 'index') {
+        setPageIndex((currentIndex) => {
+          const newArr = [...currentIndex];
+          newArr[groupingLevel] = newNumber;
+          return newArr;
+        });
+      }
 
-  useInspectButton({
-    deleteQuery,
-    loading: isLoadingGroups,
-    response,
-    setQuery,
-    refetch,
-    request,
-    uniqueQueryId,
-  });
+      if (pageType === 'size') {
+        setPageSize((currentIndex) => {
+          const newArr = [...currentIndex];
+          newArr[groupingLevel] = newNumber;
+          setStoragePageSize(newArr);
+          return newArr;
+        });
+      }
+    },
+    [setStoragePageSize]
+  );
 
-  const takeActionItems = useGroupTakeActionsItems({
-    indexName: indexPattern.title,
-    currentStatus: currentAlertStatusFilterValue,
-    showAlertStatusActions: hasIndexWrite && hasIndexMaintenance,
+  const nonGroupingFilters = useRef({
+    defaultFilters: props.defaultFilters,
+    globalFilters: props.globalFilters,
+    globalQuery: props.globalQuery,
   });
 
-  const getTakeActionItems = useCallback(
-    (groupFilters: Filter[], groupNumber: number) =>
-      takeActionItems({
-        query: getGlobalQuery([...(defaultFilters ?? []), ...groupFilters])?.filterQuery,
-        tableId,
-        groupNumber,
-        selectedGroup,
-      }),
-    [defaultFilters, getGlobalQuery, selectedGroup, tableId, takeActionItems]
-  );
+  useEffect(() => {
+    const nonGrouping = {
+      defaultFilters: props.defaultFilters,
+      globalFilters: props.globalFilters,
+      globalQuery: props.globalQuery,
+    };
+    if (!isEqual(nonGroupingFilters.current, nonGrouping)) {
+      resetAllPagination();
+      nonGroupingFilters.current = nonGrouping;
+    }
+  }, [props.defaultFilters, props.globalFilters, props.globalQuery, resetAllPagination]);
+
+  const getLevel = useCallback(
+    (level: number, selectedGroup: string, parentGroupingFilter?: string) => {
+      let rcc;
+      if (level < selectedGroups.length - 1) {
+        rcc = (groupingFilters: Filter[]) => {
+          return getLevel(
+            level + 1,
+            selectedGroups[level + 1],
+            JSON.stringify([
```

Comment:
Why do these have to be `stringified` to only be `parsed` in a recursive next step?

Reply:
If it is passed as an object it will cause unnecessary re-rendering